src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -217,7 +220,11 @@ private DocumentFileInfo MakeNonSourceFileDocumentFileInfo(MSB.Framework.ITaskIt
             var isLinked = IsDocumentLinked(documentItem);
             var isGenerated = IsDocumentGenerated(documentItem);
 
-            return new DocumentFileInfo(filePath, logicalPath, isLinked, isGenerated, SourceCodeKind.Regular);
+            var relativePath = PathUtilities.GetDirectoryName(
+                PathUtilities.GetRelativePath(_projectDirectory, filePath));
+
+            var folders = relativePath == null ? ImmutableArray<string>.Empty : relativePath.Split(PathUtilities.DirectorySeparatorChar).ToImmutableArray();
```

Comment:
Can we add some tests for this in the MSBuildWorkspace tests?

Reply:
Done